### PR TITLE
AUTO-10716: fix a funding bug in LinkAvailableBalanceMonitor

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol [hunk 1]
```diff
@@ -267,19 +266,18 @@ contract LinkAvailableBalanceMonitor is AccessControl, AutomationCompatibleInter
...
+        contractToFund.lastTopUpTimestamp + minWaitPeriod,
+        contractToFund.minBalance,
+        contractToFund.isActive
+      );
+      if (localBalance >= contractToFund.topUpAmount && fundingNeeded) {
+        bool success = i_linkToken.transfer(target, contractToFund.topUpAmount);
         if (success) {
           localBalance -= contractToFund.topUpAmount;
-          emit TopUpSucceeded(targetAddress);
+          emit TopUpSucceeded(targetAddress, contractToFund.topUpAmount); // which address should be emitted??
```
Comment (FelixFan1992) [inline contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol]: @RyanRHall @wentzeld what do u think of this address? the proxy address or the aggregator (the actual address which gets the LINK)?
and what do you think if we add the top-up amount in the event?

**Comment by wentzeld** [comment]
Some comments

File: contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol (hunk 1 above)
Comment (RyanRHall) [reply contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol]: I like both those ideas :)

File: contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol [hunk 2]
```diff
@@ -267,64 +267,65 @@ contract LinkAvailableBalanceMonitor is AccessControl, AutomationCompatibleInter
...
+      );
+      if (localBalance >= contractToFund.topUpAmount && fundingNeeded) {
+        bool success = i_linkToken.transfer(target, contractToFund.topUpAmount);
         if (success) {
           localBalance -= contractToFund.topUpAmount;
-          emit TopUpSucceeded(targetAddress);
+          emit TopUpSucceeded(target, contractToFund.topUpAmount);
         } else {
-          s_targets[targetAddress].lastTopUpTimestamp = contractToFund.lastTopUpTimestamp;
+          s_targets[targetAddress].lastTopUpTimestamp = contractToFund.lastTopUpTimestamp; // should we update this?
```
Comment (FelixFan1992) [inline contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol]: @RyanRHall  what do you think of this?

Comment (RyanRHall) [reply contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol]: this is super confusing and unnecessarily complicated lol. From what I can tell, the `lastTopUpTimestamp` is _always_ updated on line 269, and then on line 283 and 287 we set the time stamp back to what it was (meaning net no change) - is that how you interpret this code too?

I think we should take a second and think about what makes most sense from a product perspective. The min wait period is supposed to act as a rate limiter. And I think the question we're asking here is "should the rate limit apply to failed transfers?". So imagine a transfer fails for some reason, do we want the system to retry immediately or only retry after some wait period? I would think the latter, but we might want to double check with finance team.

**Review by infiloop2 — APPROVED** [review]

File: contracts/test/v0.8/automation/LinkAvailableBalanceMonitor.test.ts [hunk 3]
```diff
@@ -855,22 +878,34 @@ describe('LinkAvailableBalanceMonitor', () => {
...
+        const targetInfo1 = await labm.getAccountInfo(proxy1.address)
+        assert.notEqual(0, targetInfo1.lastTopUpTimestamp.toNumber())
         await expect(tx)
           .to.emit(labm, 'TopUpSucceeded')
-          .withArgs(proxy2.address)
+          .withArgs(aggregator2.address, twoLINK)
         await expect(tx)
           .to.emit(labm, 'TopUpSucceeded')
-          .withArgs(proxy3.address)
+          .withArgs(aggregator3.address, twoLINK)
```
Comment (RyanRHall) [inline contracts/test/v0.8/automation/LinkAvailableBalanceMonitor.test.ts:893]: 👍

**Review by RyanRHall — APPROVED** [review]
LGTM 👍

**Pull request merged**